src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -250,6 +250,17 @@ public int compare(TypedVar o1, TypedVar o2) {
       treeWalker.walkDefaultInterface((FunctionType) symbol.getType());
       emitNamespaceEnd();
     }
+    emitGoogRequireSupport(namespace, isDefault ? symbol.getName() : namespace);
+  }
+
+  private void emitGoogRequireSupport(String namespace, String closureNamespace) {
+    // goog namespace doesn't need to be goog.required.
+    if (namespace.equals("goog")) return;
+    emitNamespaceBegin("goog");
+    String qualifiedClosureNamespace = INTERNAL_NAMESPACE + "." + closureNamespace;
+    emit("function require(name: '" + closureNamespace + "'): typeof " + qualifiedClosureNamespace + ";");
```

Comment:
Might want to add a comment here pointing out the TypeScript overloading based on string arguments, otherwise this is a bit mistifying.